File: challenge_C/level2/모음사전.c
#include <stdio.h>
#include <string.h>

int ans, cnt;
char dic[6] = "AEIOU";
char res[6] = {0,};

void rec(int depth, const char *word)
{
	if(!strcmp(word, res))
	{
		ans = cnt;
		return ;
	}

	if(depth > 4)
		return ;
    
	for(int i = 0; i < 5; i++)
	{
		cnt++;
		strncat(res, &dic[i], 1);
		rec(depth + 1, word);
		res[depth] = 0;
	}
}

int solution(const char* word)
{
    rec(0, word);
    return ans;
}

File: challenge_Java/level2/모음사전.java
class Solution {
    public int wordInt = 0, answer = 0, cnt = 0;
    
    public int solution(String word) {
        for (char c : word.toCharArray()) {
            if (c == 'A') wordInt = wordInt * 10 + 1;
            else if (c == 'E') wordInt = wordInt * 10 + 2;
            else if (c == 'I') wordInt = wordInt * 10 + 3;
            else if (c == 'O') wordInt = wordInt * 10 + 4;
            else wordInt = wordInt * 10 + 5;
        }
        
        dfs(0);
        return answer;
    }

    public void dfs(int dic) {
        if (dic == wordInt) {
            answer = cnt;
            return;            
        }
        
        cnt++;
        for (int i = 1; i <= 5; i++)
            if (answer == 0 && dic / 10000 == 0)
                dfs(dic * 10 + i);
    }
}